### PR TITLE
fix null pointer exception in bn when affine is false

### PR DESCRIPTION
**Pull request by yiheng**
## What changes were proposed in this pull request?
fix null pointer exception in bn when affine is false

## How was this patch tested?
Unit tests

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/2012

### CODE REVIEW

**Comment by yiheng** [comment]
PR validation pass

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalization.scala [hunk 1]
```diff
@@ -749,6 +748,8 @@ object SpatialBatchNormalization {
     }
 
     if (scale != null) {
+      require(scale.size(1) == nChannel, "scale length is not consist with channel number")
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalization.scala]: consist => consistent

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalization.scala [hunk 2]
```diff
@@ -841,6 +841,7 @@ object SpatialBatchNormalization {
...
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalization.scala]: consist => consistent

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalization.scala [hunk 3]
```diff
@@ -896,6 +896,7 @@ object SpatialBatchNormalization {
     val n = gradOutput.nElement()
 
     if (scale != null) {
+      require(scale.size(1) == nChannel, "scale length is not consist with channel number")
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalization.scala]: consist => consistent

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalization.scala [hunk 4]
```diff
@@ -948,6 +948,7 @@ object SpatialBatchNormalization {
     var i = 0
 
     if (scale != null) {
+      require(scale.size(1) == nChannel, "scale length is not consist with channel number")
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalization.scala]: consist => consistent

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalization.scala [hunk 5]
```diff
@@ -1044,6 +1044,7 @@ object SpatialBatchNormalization {
     i = 0
     b = 0
     if (scale != null) {
+      require(scale.size(1) == nChannel, "scale length is not consist with channel number")
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalization.scala]: consist => consistent

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalization.scala [hunk 6]
```diff
@@ -1520,6 +1520,7 @@ object SpatialBatchNormalization {
     var b = 0
     var i = 0
     if (scale != null) {
+      require(scale.size(1) == nChannel, "scale length is not consist with channel number")
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalization.scala]: consist => consistent

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalization.scala [hunk 7]
```diff
@@ -1581,6 +1581,7 @@ object SpatialBatchNormalization {
...
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialBatchNormalization.scala]: consist => consistent

**Pull request merged**